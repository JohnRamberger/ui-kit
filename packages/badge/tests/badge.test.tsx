import React from "react";
import { Badge } from "../";

import { mount } from "enzyme";
import toJson from "enzyme-to-json";

const StringComponent = (): JSX.Element => {
  return <span>string</span>;
};

describe("Badge", () => {
  it("default", () => {
    const component = mount(<Badge>default</Badge>);
    expect(toJson(component)).toMatchSnapshot();
  });

  it("success", () => {
    const component = mount(<Badge appearance="success">success</Badge>);
    expect(toJson(component)).toMatchSnapshot();
  });

  it("primary", () => {
    const component = mount(<Badge appearance="primary">primary</Badge>);
    expect(toJson(component)).toMatchSnapshot();
  });

  it("danger", () => {
    const component = mount(<Badge appearance="danger">danger</Badge>);
    expect(toJson(component)).toMatchSnapshot();
  });

  it("warning", () => {
    const component = mount(<Badge appearance="warning">warning</Badge>);
    expect(toJson(component)).toMatchSnapshot();
  });

  it("outline", () => {
    const component = mount(<Badge appearance="outline">outline</Badge>);
    expect(toJson(component)).toMatchSnapshot();
  });

  it("accept jsx as children", () => {
    const component = mount(
      <Badge>
        <StringComponent />
      </Badge>
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
