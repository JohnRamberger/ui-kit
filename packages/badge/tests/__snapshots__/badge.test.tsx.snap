// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge accept jsx as children 1`] = `
<Memo()>
  <span
    className="css-iu2at0"
    data-cy="badge"
  >
    <StringComponent>
      <span>
        string
      </span>
    </StringComponent>
  </span>
</Memo()>
`;

exports[`Badge danger 1`] = `
<Memo()
  appearance="danger"
>
  <span
    className="css-1v2vcel"
    data-cy="badge"
  >
    danger
  </span>
</Memo()>
`;

exports[`Badge default 1`] = `
<Memo()>
  <span
    className="css-iu2at0"
    data-cy="badge"
  >
    default
  </span>
</Memo()>
`;

exports[`Badge outline 1`] = `
<Memo()
  appearance="outline"
>
  <span
    className="css-wrizyw"
    data-cy="badge"
  >
    outline
  </span>
</Memo()>
`;

exports[`Badge primary 1`] = `
<Memo()
  appearance="primary"
>
  <span
    className="css-1bcogwf"
    data-cy="badge"
  >
    primary
  </span>
</Memo()>
`;

exports[`Badge success 1`] = `
<Memo()
  appearance="success"
>
  <span
    className="css-1nxr7gm"
    data-cy="badge"
  >
    success
  </span>
</Memo()>
`;

exports[`Badge warning 1`] = `
<Memo()
  appearance="warning"
>
  <span
    className="css-dsjkko"
    data-cy="badge"
  >
    warning
  </span>
</Memo()>
`;
